### PR TITLE
fix(Core/SpellScript): "Egg nog" item now transforms you

### PR DESCRIPTION
**Pull request by vhiperdev**
## CHANGES PROPOSED:
-  Egg Nob - Script Item

https://classic.wowhead.com/item=17198/egg-nog#guides

you should have a chance to be transformed into [Mosshoof Runner] or [Strange Snowman]

According to "data" one of this 2 spells could be triggered upon 21149 is cast:
https://classic.wowhead.com/spell=21936/reindeer
https://classic.wowhead.com/spell=21980/snowman

https://wowwiki.fandom.com/wiki/Egg_Nog

## ISSUES ADDRESSED:
- Closes 

## TESTS PERFORMED:
-
-

## HOW TO TEST THE CHANGES:

- `.additem 17198`
- use the item and see the chances of transforming [Mosshoof Runner] or [Strange Snowman]

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1587839068992670400.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1587839068992670400');
+
+DELETE FROM `spell_script_names` WHERE `ScriptName`="spell_item_eggnog";
+INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
+(21149,'spell_item_eggnog');
```
Comment (BarbzYHOOL) [inline data/sql/updates/pending_db_world/rev_1587839068992670400.sql:5]:
```suggestion
(21149,'spell_item_eggnog');

```

**Comment by BarbzYHOOL** [comment]
correct spells, well that looks good, i just edited the title (fix in lowercase, missing ":")

**Comment by BarbzYHOOL** [comment]
but isn't it possible to do the same thing in SAI? (just being curious)

**Comment by merdlin** [comment]
Thanks, this fix really works. congratulations!

**Comment by vhiperdev** [comment]
> correct spells, well that looks good, i just edited the title (fix in lowercase, missing ":")

I don't know if it would be possible

**Comment by Kitzunu** [comment]
> but isn't it possible to do the same thing in SAI? (just being curious)

SAI for items don't exist

File: data/sql/updates/pending_db_world/rev_1587839068992670400.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1587839068992670400');
+
+DELETE FROM `spell_script_names` WHERE `ScriptName`="spell_item_eggnog";
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1587839068992670400.sql:3]: For spell_script_names always delete using both id and name. I know in this case it won't matter for this particular issue, but several spell_id's can have the same ScriptName, that is why we always should delete using both.
```suggestion
DELETE FROM `spell_script_names` WHERE `spell_id`=21149 AND `ScriptName`="spell_item_eggnog";
```

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1587839068992670400.sql:3]: an example of this is 
```sql
SELECT * FROM spell_script_names WHERE scriptname = 'spell_gen_no_offhand_proc'
``` 
It has two spell_id's with the same scriptname

Comment (vhiperdev) [reply data/sql/updates/pending_db_world/rev_1587839068992670400.sql:3]: Good observation!

File: src/server/scripts/Spells/spell_item.cpp [hunk 3]
```diff
@@ -4075,6 +4075,45 @@ class spell_item_greatmothers_soulcatcher : public SpellScriptLoader
     }
 };
 
+enum Eggnog
+{
+    SPELL_EGG_NOG_REINDEER = 21936,
+    SPELL_EGG_NOG_SNOWMAN = 21980,
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_item.cpp:4081]:
```suggestion
    SPELL_EGG_NOG_REINDEER = 21936,
    SPELL_EGG_NOG_SNOWMAN  = 21980,
```

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_item.cpp:4081]: Very cosmetic, but shouldn't lists in C++ have their "=" lined up?
Not really sure

Comment (vhiperdev) [reply src/server/scripts/Spells/spell_item.cpp]: Ty! Yes it's true updated Ty!

**Comment by pklloveyou** [comment]
Tested, no problems

**Comment by Kitzunu** [comment]
Ty for the pr

**Pull request merged**